providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -496,12 +496,14 @@ int ec_export(void *keydata, int selection, OSSL_CALLBACK *param_cb,
     if ((selection & OSSL_KEYMGMT_SELECT_OTHER_PARAMETERS) != 0)
         ok = ok && otherparams_to_params(ec, tmpl, NULL);
 
-    if (ok && (params = OSSL_PARAM_BLD_to_param(tmpl)) != NULL)
-        ok = param_cb(params, cbarg);
-    else
+    if (!ok || (params = OSSL_PARAM_BLD_to_param(tmpl)) == NULL) {
         ok = 0;
-end:
+        goto end;
+    }
+
+    param_cb(params, cbarg);
```

Comment:
why not ok = param_cb(...) ?

Reply:
oops, looks like this caused https://github.com/openssl/openssl/pull/18355#issuecomment-1152181598
since the last call-stack before the crash goes thru this line:
`#14 0x7fce64ff083d in ec_export providers/implementations/keymgmt/ec_kmgmt.c:504`

Reply:
Oops.